### PR TITLE
Fix rocksdb build on mingw.

### PR DESCRIPTION
**Pull request by vldm**

### CODE REVIEW

File: librocksdb-sys/build.rs [hunk 1]
```diff
@@ -129,6 +129,13 @@ fn build_rocksdb() {
         link("shlwapi", false);
         config.define("OS_WIN", Some("1"));
         config.define("ROCKSDB_WINDOWS_UTF8_FILENAMES", Some("1"));
+        if &target == "x86_64-pc-windows-gnu" {
+            // Tell Mingw to create localtime_r wrapper of localtime_s function.
```
Comment (vitvakatu) [inline librocksdb-sys/build.rs]:
```suggestion
            // Tell MinGW to create localtime_r wrapper of localtime_s function.
```

File: librocksdb-sys/build.rs [hunk 2]
```diff
@@ -129,6 +129,13 @@ fn build_rocksdb() {
...
+            config.define("_POSIX_C_SOURCE", None);
+            // Tell Mingw to use atleast vista headers instead of winxp.
```
Comment (vitvakatu) [inline librocksdb-sys/build.rs]:
```suggestion
            // Tell MinGW to use at least Windows Vista headers instead of the ones of Windows XP.
```

**Comment by vldm** [comment]
@aleksuss @vitvakatu CI for some reasons reports error. But i think this problem is not related to my patch. Could you try to restart job?

**Comment by vitvakatu** [comment]
Restarted, Travis does weird things lately

**Pull request merged**